pkg/monitor/monitor/util.go

```diff
@@ -688,13 +705,7 @@ func getMonitorVolumes(config *core.ConfigMap, monitor *v1alpha1.TidbMonitor) []
 		VolumeSource: core.VolumeSource{
 			ConfigMap: &core.ConfigMapVolumeSource{
 				LocalObjectReference: core.LocalObjectReference{
-					Name: config.Name,
-				},
-				Items: []core.KeyToPath{
-					{
-						Key:  "prometheus-config",
-						Path: "prometheus.yml",
```

Comment:
The original file mounted to the container is not the one that is really used by the Prometheus, please check the code here https://github.com/pingcap/tidb-operator/blob/master/pkg/monitor/monitor/util.go#L415, how would you like to watch the changes here and notify the Prom to use the latest config?

Reply:
This pr just separate grafana configuration from the old config, the next pr i will use promtheus reloader replace the design of prometheus started by  sed command.

Refer:https://github.com/thanos-io/thanos/blob/main/docs/components/sidecar.md  `reloader.config-envsubst-file`,this configuration will be decompressed if needed, environment variables, it will be substituted and the output written into the given path .
![image](https://user-images.githubusercontent.com/13257127/128967856-6e1e2ab5-9d87-4b14-b9e2-de30b3afd245.png)